### DISCUSSION TITLE
欢迎来带obsidian-weread-plugin 讨论组!

### DISCUSSION THREAD

**Discussion by zhaohongxuan** [zhaohongxuan/obsidian-weread-plugin#30] [Announcements]
## 👋 欢迎使用微信读书Obsidian插件!

这里可以讨论关于Obsidian 微信读书插件相关的一些问题和建议，使用过程中有疑问或者有更好的`模板`都可以在这里分享出来。在这里你可以：

* 提问你感到困惑的问题
* 分享你的使用插件以及Obsidian的一些技巧和创意
* 在这里只需要敞开心扉讨论就好💪.

**Comment by yoto-Github** [comment]
obsidian版本0.14.15
安装了最新的weread-plugin，但无论如何同步微信读书都只有4本书能成功同步进来
而这4本的内容更新同步都是正常的

**Reply by zhaohongxuan** [reply]
我使用最新的0.2.3本地没有办法重现，也是最新的0.14.15，你可以打开console看下日志，看是否有报错的信息。
<img width="719" alt="image" src="https://user-images.githubusercontent.com/8613196/173714760-c031adc8-42ef-4335-b0f6-1e5c00a5618a.png">

**Comment by zhaohongxuan** [comment]
把notecount限制去掉后，重现bug了，已经在0.2.5版本fix了

**Reply by yoto-Github** [reply]
新版本修复了~感谢

**Comment by physicsok** [comment]
请问为什么我一扫完码obsidian就闪退了

**Reply by zhaohongxuan** [reply]
新建一个vault，单独安装一个插件试试，可能是插件冲突。

**Comment by Haxxu6** [comment]
大佬！求助，我正常下载安装总显示失败
![image](https://user-images.githubusercontent.com/108344897/176184671-442ac8a0-5fdc-4662-9945-6ea4f181fe44.png)

**Reply by zhaohongxuan** [reply]
打开控制台贴一下 log

**Comment by huiyin7** [comment]
使用dataview 添加元数据字段是在设置插件里面去添加还是一个个笔记的元数据里面手动加呢，刚使用不久不是很懂，麻烦请教

**Reply by zhaohongxuan** [reply]
> 使用dataview 添加元数据字段是在设置插件里面去添加还是一个个笔记的元数据里面手动加呢，刚使用不久不是很懂，麻烦请教

直接在笔记的frontmatter里面贴就好了，yaml格式的

**Comment by Arcopsyder** [comment]
看到使用说明中："⚠️ 本插件是覆盖式更新，请不要在同步的文件里修改内容，写永久笔记（为什么写永久笔记参考[《卡片笔记写作法》](https://book.douban.com/subject/35503571/)）的时候可以使用[Block引用](https://help.obsidian.md/How+to/Link+to+blocks) 的方式，在外部引用进行批注。" 有两个疑问。  
第一个问题是，意思是可以在同步的文件中创建块么？（但是创建块不也是修改了md文件的内容么？）
第二个疑问是，如果我在同步的文件中修改了内容会发生什么事情？以后同步功能都会故障，还是说就算我修改了，同步微信读书以后也会把我的修改给覆盖掉，如果是会覆盖掉，那我创建的块不也会被覆盖掉么？

**Reply by zhaohongxuan** [reply]
block是自动生成的，不能自己修改。

**Comment by sorzero79** [comment]
太好用了，支持作者！ 我在每条评论前加上 #读书心得，就可以通过标签筛选，简直不要太方便！

**Comment by amelloly** [comment]
标注可以支持区分是波浪线、马克笔还是直线么？看书的时候，我会通过这三种不同的标注类型来区分划线目的

**Comment by Dennifer** [comment]
这个插件实在太爱了！期待作者可以持续维护并完善~ 如果考虑到时间成本问题，也可以发一个打赏二维码，真切需要的朋友们（比如我）肯定很愿意做出一些些贡献的！

**Comment by Zhanglela** [comment]
请教一下怎么修改默认模板呢？

**Comment by Zouz-code** [comment]
大佬这个报错怎么解决呀

`同步微信读书笔记异常 TypeError: Cannot read properties of undefined (reading 'replace')
    at eval (plugin:obsidian-weread-plugin:2:315813)
    at Array.map (<anonymous>)
    at eval (plugin:obsidian-weread-plugin:2:315387)
    at E.eval (plugin:obsidian-weread-plugin:2:315853)
    at Generator.next (<anonymous>)
    at a (plugin:obsidian-weread-plugin:2:303917)
eval @ plugin:obsidian-weread-plugin:2
s @ plugin:obsidian-weread-plugin:2
Promise.then（异步）
l @ plugin:obsidian-weread-plugin:2
eval @ plugin:obsidian-weread-plugin:2
e @ plugin:obsidian-weread-plugin:2
startSync @ plugin:obsidian-weread-plugin:2
eval @ plugin:obsidian-weread-plugin:2
’

**Reply by zdz316** [reply]
我遇到同样的问题，请问解决了吗？

**Reply by Zouz-code** [reply]
没有呢，这个插件没用成功。。无奈😔

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Reply by Songyayun** [reply]
您好，我不小心把插件删除了，重新安装后就无法同步笔记了，然后总黑屏，不知道怎么办？谢谢啦~

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Reply by Songyayun** [reply]
太谢谢您了，我开始安装挺好的，我同步一下把我微信几百本书都同步了，信息量太大，图谱哪里很庞大都是独立的，我觉得不太好，然后我删除笔记，就把微信读书里面几百本书都删除，只保留几十本重新同步，然后发现同步不了，然后我删除插件重装，现在就是0更新。

**Comment by muyajun** [comment]
ipad，找不到微信登陆的模块，请指点

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by ylzty** [comment]
你好，现在微信读书网页版已经加入五种颜色、三种样式的划线，在后续开发中能不能使下划线样式也显示在笔记中呢？因为我在阅读的过程中常常使用不同样式的下划线来表示书摘的不同重要程度，没有这个我在obsidian里看得不太方便。感谢！

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by ylzty** [comment]
噢，还有一个，在文件夹分类里是否可以加上“按照首字母排序分类”或者“按照自定义分组分类”呢？

**Comment by xieyingli** [comment]
这两天冒出来一个竞争者了，notepal，不过还是感觉你这个好用

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by ZhaoRanO** [comment]
求助，我遇到一个问题，在我登录后，不能完成同步，显示cookie失效
<img width="1018" alt="Screenshot 2023-08-07 at 10 52 48 AM" src="https://github.com/zhaohongxuan/obsidian-weread-plugin/assets/32949813/17414eb8-f1d4-4043-baf8-bb52db479ab6">

**Reply by ZhaoRanO** [reply]
版本是0.5.2

**Reply by ZhaoRanO** [reply]
我重新安装了0.4.10版本，又能重新同步了

**Reply by Kairitsuhou** [reply]
同款问题，cy

**Comment by youyou2270** [comment]
很棒的插件～ 不知道能否实现高亮不同颜色之间的区分以及搜索？？谢谢

**Reply by Moyf** [reply]
我做了一个 [模板](https://github.com/zhaohongxuan/obsidian-weread-plugin/discussions/62#:~:text=%E9%98%B6%E6%94%B9%E5%8A%A8-,%E9%AB%98%E4%BA%AE%E9%A2%9C%E8%89%B2%E7%9A%84%E4%B8%8D%E5%90%8C%E6%A0%B7%E5%BC%8F,-%E9%80%82%E9%85%8D%E4%BA%86) 支持颜色区分，可以参考看看~

**Reply by youyou2270** [reply]
非常感谢，我下载了新的版本0.5.5，并把设置那里换成你发给我的网页中的模板（复制了代码），也重新启动了OB，但是似乎没有变化，并且也不能实时更新在微信读书中新增加的笔记了。不知道是我操作的问题吗？

**Reply by Moyf** [reply]
咦…… 我刚又测试了一下，没有问题耶。

你可以按 `Ctrl+Shift+I` 打开控制台窗口，看一下有没有红字的报错提示吗？
![image](https://github.com/zhaohongxuan/obsidian-weread-plugin/assets/4013062/9729540e-22fe-4dcf-a21e-b92e3925cb2b)

**Reply by youyou2270** [reply]
模版不错～ 换了最新的插件版本，已经解决了～ 谢谢

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by unique-space** [comment]
公众号划线和笔记归类同步是如何实现的？

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by YeChen123e** [comment]
您好，首先很感谢您开发了这个插件，但是我在使用发现划线或者评论的时间和实际时间有出入，请问怎么解决呀

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by zhchh008** [comment]
何时能够实现反向导入（导出的笔记导入到对应的epub文件中）？

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by ijianjiandan** [comment]
您好，请问一下，微信读书里公众号的文章的标注不能同步到OB，只看到书籍标注的同步，请问是什么原因？另外同步的时候提示有9本书同步，但是OB上的目录只有8本,不知道是不是这个新的公众号文章标注的1没有同步过来?谢谢

**Comment by LiuPiii** [comment]
您好，请问微信读书里公众号的文章和标注同步到OB是如何实现的？谢谢

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by 42665286** [comment]
求助，更换微信读书账号后登录问题。
 之前的账号退出登录之后，现在的账号怎么都登不上去。明明扫码登录完成但仍显示登录按钮，点进登录按钮，里面是已经登录的账号。点击同步，报错说cookie失效，请重新扫码登录。
![屏幕截图(1211)](https://github.com/user-attachments/assets/47cdf135-21f1-4355-a42a-a8d399de191a)

**Reply by 42665286** [reply]
版本是0.11.0

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by youyou2270** [comment]
求助，很久没同步ob中的微信读书，今天想同步下，就报错了，如图。插件版本0.11.2，OB是最新版本1.8.4（ob版本更新前就发现报错，所以更新了ob版本到最新）
<img width="545" alt="截屏2025-02-04 14 47 33" src="https://github.com/user-attachments/assets/e0e9af5b-0540-4ff6-9004-dd5163c77248" />
<img width="241" alt="截屏2025-02-04 14 47 40" src="https://github.com/user-attachments/assets/a67c3841-18c8-4385-9107-274941224b8a" />

**Reply by youyou2270** [reply]
重新扫码登陆也不行

**Reply by youyou2270** [reply]
已解决，重新登陆cookie后，还需要点击一下自己的微信阅读的头像，就可以刷新了，之后可以正常同步。

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。

**Comment by Ych4321** [comment]
作者你好，非常感谢制作了这款高效的插件了。我想问一下可否增加一个功能只同步已经阅读完成的书籍笔记呢？比如说以是否读完作为标记，或者只同步微信读书中的某个分组？ 谢谢！因为我每次会一次性导入大量书籍，但是大部分书没有读完，笔记就会非常凌乱。

**Comment by Zhanglela** [comment]
您好，张乐拉已经收到了您的邮件，稍后我将尽快给您回复。
